tests/test_action.py

```diff
@@ -45,3 +46,44 @@ class TestAction(Action):
 
     with pytest.raises(TypeError):
         TestAction()
+
+
+def test_accept_method_calls_visit_and_response_and_requests_accept():
+    mock_wrapper = Mock()
+    mock_wrapper.mocked_visitor = MagicMock()
+    mock_wrapper.mocked_request = MagicMock()
+    mock_wrapper.mocked_response1 = MagicMock()
+    mock_wrapper.mocked_response2 = MagicMock()
```

Comment:
instead of using magicmock, wouldn't be better to create a TestVisitor class that just reproduces each element as called in `accept()` ?

Reply:
Thanks a lot @daneoshiga. I will implement.

Reply:
@daneoshiga I implemented one complete test for the visitor itself using the a `TestVisitor`. Take a look and tell me what you think.

I don't think the `accept` tests with mocks are needed anymore. Just need a second thought.

Reply:
I think it's clearer but also think ths test is not that much _ unitary_.

The order on multiple ~chldren~ elements doesn't matter. It just matter that they are being visited right after the parent element.

Reply:
you can split it in multiple ones, like `test_visitor_api_call_order` and pass an `API` object with just API and Version element.. than the same for version and route, then route, action and querystring...
that way it's gonna be more unitary, (not sure the fixtures created will be so useful then, so maybe each API will need to be created inside the test, which isn't a bad thing)

Reply:
done, you can review it again.